### PR TITLE
[ntuple] outline several definitions from RFieldBase

### PR DESCRIPTION
**Pull request by silverweed**
# This Pull request:
moves various definitions outside of `RFieldBase` for improved readability

## Checklist:

- [x] tested changes locally
- [ ] updated the docs (if necessary)

### CODE REVIEW

**Review by jblomer — APPROVED** [review]
In principle fine, but I think at least for the classes, I'd prefer to move the docstring as well to where the actual code is.

File: tree/ntuple/v7/inc/ROOT/RFieldBase.hxx [hunk 1]
```diff
@@ -500,50 +360,17 @@ protected:
...
-
-      if (fTraits & kTraitMappable)
-         fPrincipalColumn->Read(clusterIndex, to);
-      else
-         ReadInClusterImpl(clusterIndex, to);
-      if (R__unlikely(!fReadCallbacks.empty()))
-         InvokeReadCallbacks(to);
-   }
+   void Read(NTupleSize_t globalIndex, void *to);
+   void Read(RClusterIndex clusterIndex, void *to);
```
Comment (hahnjo) [inline tree/ntuple/v7/inc/ROOT/RFieldBase.hxx:373]: Outlining `Append` and `Read` does not allow the compiler to inline them anymore. For simple fields, this inlining allows to directly call into the (principal) `RColumn`. I'm not sure if this is important, but it's also not clear to me why these functions have to be outlined? They don't seem to use any of the outlined `class`es / `struct`s...

Comment (silverweed) [reply tree/ntuple/v7/inc/ROOT/RFieldBase.hxx:373]: No reason other than "they take space". If it's important for them to be inlined for performance reasons then I'll just put them back there

Comment (jblomer) [reply tree/ntuple/v7/inc/ROOT/RFieldBase.hxx]: Good point! Yes, these two methods should remain inline-able.

**Pull request merged**